### PR TITLE
Create responsive stacked table

### LINKED ISSUE #104: Create new table variation
## Issue
We need to re-create the responsive table variation used on the buses page.

## Desktop

Image

## Mobile

Image

### PR DESCRIPTION
**Pull request by maxatdetroit**
## Closes #104 

This PR:
* Renames 'responsive' table to 'scrollable' since we now have multiple responsive table styles
* Introduces two stacked, responsive table styles into the UXDS:
  1. Blocker header
  2. Inline header

**Note**: I have this merging into fix.table because I'm relying on table HTML elements for styling in this PR.

#### Before

Screenshot 2023-10-24 at 12 41 33 PM
Screenshot 2023-10-24 at 12 41 41 PM

#### After

Screenshot 2023-10-24 at 12 42 13 PM
Screenshot 2023-10-24 at 12 42 19 PM

#### Large Viewport (Block Header)

Screenshot 2023-10-30 at 6 26 35 PM

#### Small Viewport (Block Header)

Screenshot 2023-10-30 at 6 24 46 PM

#### Large Viewport (Inline Header)

Screenshot 2023-10-30 at 6 26 35 PM

#### Small Viewport (Inline Header)

Screenshot 2023-10-30 at 6 24 34 PM

### CODE REVIEW

**Comment by bankovic** [comment]
Maybe help differentiating between the rows on phones by 5px left border and then use them in striped fashion - odd/even rows. Along these lines: https://detroitmi.gov/departments/buildings-safety-engineering-and-environmental-department/bseed-divisions/development-resource-center/quick-start-guide-new-business

File: src/stories/table.stories.js [hunk 1]
```diff
@@ -341,3 +341,54 @@ export const Responsive = () => html`
...
+        <cod-table-cell data-label="Bus Route Name">Grand River</cod-table-cell>
+        <cod-table-cell data-label="Route Viewer">
+          <a href="https://example.com">View Route</a>
+        </cod-table-cell>
+        <cod-table-cell data-label="Downloadable Schedule">
+          <a href="https://example.com">Download Schedule</a>
+        </cod-table-cell>
+      </cod-table-row>
+    </cod-table-body>
+  </cod-table>
```
Comment (maxatdetroit) [inline src/stories/table.stories.js]: @sreidthomas these are the HTML tags that we'll use for the bus schedule view table once this PR gets merged in.

**Comment by maxatdetroit** [comment]
Here's what the table looks like with zebra-striped left borders:

Screenshot 2023-10-30 at 12 46 42 PM

I also changed the tint/shade of the borders and backgrounds to add more contrast.

The header background makes the table pretty hard to read at-a-glance since they have a zebra-stripe effect too.

Frankly, if we're intent on keeping the header background color (which I agree makes the table more versatile since it will support bolded cells easily), then I feel we should go back to the original stacked style:

image

**Comment by bankovic** [comment]
[like]  Goran Bankovic reacted to your message:
________________________________
From: Max Morgan ***@***.***>
Sent: Monday, October 30, 2023 4:50:36 PM
To: CityOfDetroit/COD-Design-System ***@***.***>
Cc: Goran Bankovic ***@***.***>; Review requested ***@***.***>
Subject: [EXTERNAL] Re: [CityOfDetroit/COD-Design-System] Create responsive stacked table (PR #112)


Here's what the table looks like with zebra-striped left borders:

[Screenshot 2023-10-30 at 12 46 42?PM]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/143553259/279133946-9faf785d-8468-42ad-8e13-f74a92225077.png__;!!NcXZSU8rfchoEksI!Y9rsEfKSqEwzQNuSkPmpEJ1JtZhssE8id69pNLK5EbycNecFcLAAr7xx8uEw5cNXb1zMfDZvuUOgKL-W6BWFnqtrIyptY9c$>

I also changed the tint/shade of the borders and backgrounds to add more contrast.

The header background makes the table pretty hard to read at-a-glance since they have a zebra-stripe effect too.

Frankly, if we're intent on keeping the header background color (which I agree makes the table more versatile since it will support bolded cells easily), then I feel we should go back to the original stacked style:

[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/143553259/279134279-301e5840-91fd-4874-b2da-c14cae27d9f1.png__;!!NcXZSU8rfchoEksI!Y9rsEfKSqEwzQNuSkPmpEJ1JtZhssE8id69pNLK5EbycNecFcLAAr7xx8uEw5cNXb1zMfDZvuUOgKL-W6BWFnqtrlxRf4I0$>

**Comment by maxatdetroit** [comment]
Accidentally closed the PR. 🐙

**Comment by maxatdetroit** [comment]
I've updated the PR to support both inline and block header styles as discussed in person this afternoon.

#### Large Viewport (Both Styles)

Screenshot 2023-10-30 at 6 26 35 PM

#### Small Viewport (Inline Header)

Screenshot 2023-10-30 at 6 24 34 PM

#### Small Viewport (Block Header)

Screenshot 2023-10-30 at 6 24 46 PM

**Comment by bankovic** [comment]
mobile-table-1
mobile-table-2 1
mobile-table-2
mobile-table-3 1
mobile-table-3
Here are few versions with borders.

**Comment by maxatdetroit** [comment]
I think the multi-color border solution avoids the underlying problem which is that the block header layout doesn't work well with multiple header per row. There's just too much going on when there are multiple header. I think we should keep the single-color zebra striped border and restrict the usage of the block header layout to tables where only a single header is needed. The bus route table is actually a good example of this:

Screenshot 2023-10-31 at 12 15 03 PM

**Review by jedgar1mx — COMMENTED** [review]
I see there is some change in naming convention for some of the files as well as some new scss files. Is there a reason for the new extra files?

**Comment by maxatdetroit** [comment]
The new extra files are SASS files used to generate the stylesheets for the table family of components. We have to use SASS for those (unlike the CSS component style sheets) because we need to use the bootstrap media breakpoint utilities and variables. 

Currently, the table-specific SASS-generated CSS is added to the end of the themed bootstrap CSS which is inlined into every single component. If we wanted to keep the stylesheets more encapsulated with their components, we can update the SASS build script (`compile_bootstrap.sh`) to output the table-specific SASS-generated CSS into the relevant table component folders and continue using those as inline stylesheets on the component.

**Review by jedgar1mx — APPROVED** [review]
Ok, we can look into improving the sass process as a separate task.

**Comment by maxatdetroit** [comment]
Created a new issue for the SASS compilation and encapsulation: https://github.com/CityOfDetroit/COD-Design-System/issues/115.

**Pull request merged**